### PR TITLE
Respect specification defaults.

### PR DESCRIPTION
**Pull request by andrewchambers**

### CODE REVIEW

File: src/fspec-archive.c [hunk 1]
```diff
@@ -41,8 +41,23 @@ filetype(const char *type)
...
+        return 0755;
+    if (strcmp(type, "sym") == 0)
+        return 0777;
+    fprintf(stderr, "unknown file type '%s'\n", type);
+    exit(1);
+}
+
 int main(int argc, char **argv)
 {
+    int set_default_mode = 1;
```
Comment (michaelforney) [inline src/fspec-archive.c]: I think this needs to go at the top of the do-loop, since it's per-entry.

File: src/fspec-archive.c [hunk 2]
```diff
@@ -41,11 +41,26 @@ filetype(const char *type)
...
+    if (strcmp(type, "sym") == 0)
+        return 0777;
+    fprintf(stderr, "unknown file type '%s'\n", type);
+    exit(1);
+}
+
 int main(int argc, char **argv)
 {
-    char *line = NULL;
-    size_t len = 0;
```
Comment (michaelforney) [inline src/fspec-archive.c:47]: We actually need these initializations for the first call to getline (so it knows nothing was previously allocated).

File: src/fspec-archive.c [hunk 3]
```diff
@@ -80,9 +98,15 @@ int main(int argc, char **argv)
...
+                set_default_mode = 0;
+                mode = strtol(line + 5, NULL, 8);
             } else if (strncmp(line, "type=", 5) == 0) {
-                archive_entry_set_filetype(entry, filetype(line + 5));
+                const char *t = line + 5;
+
+                archive_entry_set_filetype(entry, filetype(t));
+                if (set_default_mode) {
+                    mode = defaultmode(t);
+                }
```
Comment (michaelforney) [inline src/fspec-archive.c:108]: I think this should be moved to after the attribute parsing loop, so that `mode` and `type` could appear in either order.

Comment (michaelforney) [reply src/fspec-archive.c:108]: Err... never mind, how it is currently is fine.

File: src/fspec-archive.c (hunk 2 above)
Comment (andrewchambers) [reply src/fspec-archive.c:47]: Fuck Im sorry, what a fail.

Comment (michaelforney) [reply src/fspec-archive.c:47]: No worries, that's what reviews are for.

**Review by michaelforney — APPROVED** [review]

**Pull request merged**